bootstore/src/schemes/v0/peer.rs

```diff
@@ -971,4 +1226,176 @@ mod tests {
+        for handle in [&learner_handle, &handle0, &handle1, &handle2] {
+            let _ = handle.load_peer_addresses(addrs.clone()).await;
+        }
+
+        // Poll the learner to ensure it gets the network config
+        // Note that the learner doesn't even need to learn its share
+        // for network config replication to work.
+        let start = Instant::now();
+        let mut done = false;
+        while !done {
+            let timeout = POLL_TIMEOUT.saturating_sub(Instant::now() - start);
+            tokio::select! {
+                _ = sleep(timeout) => {
+                    panic!("Network config not replicated");
+                }
+                res = learner_handle.get_network_config() => {
+                    if res.unwrap().as_ref() == Some(&network_config) {
+                        done = true;
+                    }
+                }
+            }
+        }
+
+        // Stop node0, bring it back online and ensure it still sees the config
+        // at generation 1
+        handle0.shutdown().await.unwrap();
+        jh0.await.unwrap();
+        let (mut node0, handle0) = Node::new(config[0].clone(), &log).await;
+        let jh0 = tokio::spawn(async move {
+            node0.run().await;
+        });
+        assert_eq!(
+            Some(&network_config),
+            handle0.get_network_config().await.unwrap().as_ref()
+        );
+
+        // Stop node0 again, update network config via node1, bring node0 back online,
+        // and ensure all nodes see the latest configuration.
+        let new_config = NetworkConfig {
+            generation: 2,
+            blob: b"Some more network data".to_vec(),
+        };
+        handle0.shutdown().await.unwrap();
+        jh0.await.unwrap();
+        handle1.update_network_config(new_config.clone()).await.unwrap();
+        assert_eq!(
+            Some(&new_config),
+            handle1.get_network_config().await.unwrap().as_ref()
+        );
+        let (mut node0, handle0) = Node::new(config[0].clone(), &log).await;
+        let jh0 = tokio::spawn(async move {
+            node0.run().await;
+        });
+        let start = Instant::now();
+        // These should all resolve instantly, so no real need for a select,
+        // which is getting tedious.
+        // We also want to repeatedly loop until all consistently have the same version
+        // to give some assurance that the old version from node0 doesn't replicate
+        'outer: loop {
+            if Instant::now() - start > POLL_TIMEOUT {
+                panic!("network config not replicated");
+            }
+            for h in [&handle0, &handle1, &handle2, &learner_handle] {
+                if h.get_network_config().await.unwrap().as_ref()
+                    != Some(&new_config)
+                {
+                    // We need to try again
+                    continue 'outer;
+                }
+            }
+            // Success
+            break;
+        }
```

Comment:
```suggestion
        loop {
            if Instant::now() - start > POLL_TIMEOUT {
                panic!("network config not replicated");
            }
            if [&handle0, &handle1, &handle2, &learner_handle].into_iter().all(
                |h| {
                    h.get_network_config().await.unwrap().as_ref()
                        == Some(&new_config)
                },
            ) {
                break;
            }
        }
```

Reply:
I like it!

Reply:
Hmm, this actually doesn't work because of the `.await` inside the closure. I get the following compilation error:

```
   Compiling bootstore v0.1.0 (/home/ajs/omicron/bootstore)
error[E0728]: `await` is only allowed inside `async` functions and blocks
    --> bootstore/src/schemes/v0/peer.rs:1354:43
     |
1353 |                 |h| {
     |                 --- this is not `async`
1354 |                     h.get_network_config().await.unwrap().as_ref()
     |                                           ^^^^^^ only allowed inside `async` functions and blocks
```

Reply:
Ugh, I hate it when that happens.  It looks like you may be able to use `join_all`, but I'm fine skipping the cleanup if it's too hard.

Reply:
Just bigger fish to fry. Got a marlin and a 55 gallon drum of soybean oil!